packages/contracts-bedrock/src/L2/L2StandardBridgeInterop.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L2StandardBridge } from "./L2StandardBridge.sol";
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+/// @notice Thrown when the decimals of the tokens are not the same.
+error InvalidDecimals();
+
+/// @notice Thrown when the legacy address is not found in the OptimismMintableERC20Factory.
+error InvalidLegacyAddress();
+
+/// @notice Thrown when the SuperchainERC20 address is not found in the SuperchainERC20Factory.
+error InvalidSuperchainAddress();
+
+/// @notice Thrown when the remote addresses of the tokens are not the same.
+error InvalidTokenPair();
+
+// TODO: Use OptimismMintableERC20Factory contract instead of interface
+interface IOptimismMintableERC20Factory {
+    function deployments(address) external view returns (address);
+}
+
```

Comment:
We could use an interface that is generic with just the `deployments` function. We currently do this sometimes, it reduces the number of files when verifying a contract and i think it will reduce compile time if we use interfaces everywhere rather than importing full implementations